src/http/validations/infusion.js

```diff
@@ -0,0 +1,22 @@
+import FieldErrors from './fieldErrors';
+import db from '../../db';
+
+const createInfusion = (req, res, next) => {
+  const {
+    patientName, doctorsInstruction, startVolume, stopVolume, deviceId,
+  } = req.body;
+  const fieldErrors = new FieldErrors();
+
+  if (!patientName || typeof (patientName) !== 'string') fieldErrors.addError('patientName', 'patientName is required');
+  if (!doctorsInstruction || typeof (doctorsInstruction) !== 'string') fieldErrors.addError('doctorsInstruction', 'doctorsInstruction is required');
+  if (!startVolume) fieldErrors.addError('startVolume', 'startVolume is required');
+  if (!stopVolume) fieldErrors.addError('stopVolume', 'stopVolume is required');
+  if (!db.validResourceId(deviceId)) fieldErrors.addError('deviceId', 'Invalid deviceId.');
+  if (!deviceId || typeof (deviceId) !== 'string') fieldErrors.addError('deviceId', 'deviceId is required');
```

Comment:
Validating that the deviceId is given should come before validating that it's a valid resourceId

Reply:
fixed